### ISSUE TITLE
MissingFilter with Sort errored with json integer overflow detected

### ISSUE THREAD

**Issue by alexander-schranz** [ongr-io/ElasticsearchBundle#397]
```
    $rangeFilter = new RangeFilter($field, $value);
    $missingFilter = new MissingFilter($field); // missing filter combined with sort will errored
    $orFilter = new OrFilter(array($rangeFilter, $missingFilter));
    $search->addFilter($orFilter);
    $sort = new Sort($sortField, $sortType);
    $search->addSort($sort); // sort combined with missing filter will errored
    $result = $this->getRepository()->execute($search);
```

Notice: json_decode(): integer overflow detected

Googling for the error seems that there is possible to fix it by add `'missing' => PHP_INT_MAX -1,` for the sort but don't know where I can add this in ongr.

**Comment by GrandLTU** [comment]
Tried to replicate - seems to work fine.

``` php
<?php

/*
 * This file is part of the ONGR package.
 *
 * (c) NFQ Technologies UAB <info@nfq.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace ONGR\ElasticsearchBundle\Tests\Functional\DSL\Complex;

use ONGR\ElasticsearchBundle\DSL\Filter\MissingFilter;
use ONGR\ElasticsearchBundle\DSL\Filter\OrFilter;
use ONGR\ElasticsearchBundle\DSL\Filter\RangeFilter;
use ONGR\ElasticsearchBundle\DSL\Sort\Sort;
use ONGR\ElasticsearchBundle\Test\AbstractElasticsearchTestCase;

/**
 * Class RangeFilterAndMissingFilterAndSortTest.
 */
class RangeFilterAndMissingFilterAndSortTest extends AbstractElasticsearchTestCase
{

    /**
     * {@inheritdoc}
     */
    protected function getDataArray()
    {
        return [
            'default' => [
                'product' => [
                    [
                        '_id' => 1,
                        'title' => 'baz',
                        'price' => 144,
                        'description' => 'Lorem ipsum',
                    ],
                    [
                        '_id' => 2,
                        'title' => 'bar',
                        'price' => 256,
                        'description' => 'Lorem ipsum',
                    ],
                    [
                        '_id' => 3,
                        'title' => 'baz',
                        'price' => 260,
                        'description' => 'Lorem ipsum',
                    ],
                    [
                        '_id' => 4,
                        'title' => 'foo',
                        'price' => 275,
                        'description' => 'Lorem ipsum',
                    ],
                    [
                        '_id' => 5,
                        'title' => 'foo',
                        'price' => 276,
                        'description' => 'Lorem ipsum',
                    ],
                    [
                        '_id' => 6,
                        'title' => 'baz',
                        'price' => 1000,
                        'description' => 'Lorem ipsum',
                    ],
                    [
                        '_id' => 7,
                        'title' => 'baz',
                        'description' => 'Lorem ipsum',
                    ],
                ],
            ],
        ];
    }

    /**
     * Test filters in aggregations and post filter.
     */
    public function testRangeFilterAndMissingFilterAndSort()
    {
        $repository = $this->getManager()->getRepository('AcmeTestBundle:Product');
        $search = $repository->createSearch();

        $rangeFilter = new RangeFilter('price', ['gte' => 0, 'lte' => 275]);
        $missingFilter = new MissingFilter('price');
        $orFilter = new OrFilter([$rangeFilter, $missingFilter]);
        $search->addFilter($orFilter);
        $sort = new Sort('price');
        $search->addSort($sort);
        $result = $repository->execute($search);
    }
}
```

**Comment by alexander-schranz** [comment]
I'm on symfony 2.6 on version 0.9.1
the error is thrown in ElasticSearch\Serializers\SmartSerializer in the jsonDecode

the json response  have this in it: 

```
"sort":[9223372036854775807]
```

**Comment by saimaz** [comment]
@alexander-schranz can you write a test case to reproduce a bug you found? Take as an example @aivsev pasted code snippet. Right now we cannot reproduce it with Symfony 2.6 and ESB 0.9.1

**Comment by alexander-schranz** [comment]
seems the error is not part of ongr bundle so I will close this issue

The error itself is thrown in \Elasticsearch\Serializers\AbstractJsonSerilizer:jsonDecode:L44

What I found about the error is this PR in elasticsearch to throw exception when this occurs
https://github.com/elastic/elasticsearch-php/pull/146 in my configuration it will not throw an exception because of the notice json_decode give. On the production system it currently works so its ok for me now because its only one development system where this error occurs.

Also an issue I found in the json pecl extension for this error: https://github.com/remicollet/pecl-json-c/issues/13 maybe it have todo with the installed version of pecl on the errored system I have 1.3.6.
